### DISCUSSION TITLE
Missing properties.*.values objects for first few hours sometimes: suspect previous/unreported duration declaration

### DISCUSSION THREAD

**Discussion by alexmkio** [weather-gov/api#443] [General]
I am attempting to filter for several variables (temp, wind speed, and probability of precipitation) based off of selected user thresholds. I am running into an issue where sometimes the data is undefined for the first n number of hours for a certain variable.

I understand the [duration](https://en.m.wikipedia.org/wiki/ISO_8601#Durations) format and its implications. I am currently controlling for when H is found to be greater than 1.

My guess as to why some time periods still have no data, and why they only ever seem to be in the beginning of the array, is because a previous duration greater than 1H was declared/reported prompting for an "absence" of explicit data for the next n  number of hours. Ex: a 4 hour duration for a property value was declared at 11:00, but the fetch happens at 12:00. No value is reported until 15:00, and the user has no way of knowing what the previous value was in order to fill the gap. 

Thoughts? Am I missing something?

**Comment by alexmkio** [comment]
Here is some data from a fetch I made yesterday:

```
"temperature": {
  "uom": "wmoUnit:degC",
  "values": [
    {
      "validTime": "2021-08-04T06:00:00+00:00/PT3H",
      "value": 23.333333333333332
    },
```
    
```
"windSpeed": {
  "uom": "wmoUnit:km_h-1",
  "values": [
    {
      "validTime": "2021-08-04T06:00:00+00:00/PT1H",
      "value": 7.408
    },
```
    
```
"probabilityOfPrecipitation": {
  "uom": "wmoUnit:percent",
  "values": [
    {
      "validTime": "2021-08-04T10:00:00+00:00/PT1H",
      "value": 59
    },
```
    
probabilityOfPrecipitation starts defining values at 10:00, whereas the other two start at 6:00 (when the fetch originated).

**Reply by jah1007** [reply]
Could you provide the URL you are using? As each Weather Forecast Office (WFO) creates their own data, of which there is no real standard around, local variations are possible.

**Reply by alexmkio** [reply]
https://api.weather.gov/gridpoints/JAX/40,29
